File: problems/1013-partition-array-into-three-parts-with-equal-sum/solution.py
class Solution:
    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
        total_sum = sum(arr)
        if total_sum % 3 != 0:
            return False
        target_sum = total_sum // 3
        current_sum = 0
        count = 0
        for num in arr:
            current_sum += num
            if current_sum == target_sum:
                count += 1
                current_sum = 0
        return count >= 3

File: problems/1013-partition-array-into-three-parts-with-equal-sum/solution.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& arr) {
        int sum = 0;
        for (int num : arr) {
            sum += num;
        }

        if (sum % 3 != 0) {
            return false;
        }

        int target = sum / 3;
        int currentSum = 0;
        int count = 0;

        for (int i = 0; i < arr.size(); i++) {
            currentSum += arr[i];
            if (currentSum == target) {
                count++;
                currentSum = 0;
            }
        }

        return count >= 3;
    }
};